### PR TITLE
Add Typst details to main citation page

### PR DESCRIPTION
**Pull request by cwickham**
[Preview](https://deploy-preview-1209.quarto.org/docs/authoring/citations.html)
Closes https://github.com/quarto-dev/quarto-cli/issues/9762

### CODE REVIEW

**Comment by cwickham** [comment]
@cderv This adds the Typst details to the main citation page and calls out that Typst is different at the top.  I think it still may be confusing since it isn't clear what parts of the page still apply to Typst (I think none, except "Citation Syntax" and I guess "Bibliography Files" although the supported formats might be different). Is it worth a bigger re-organize to clear that up, or is what is in this PR good enough.

File: docs/output-formats/_typst-citations.md
Comment (cderv) [inline docs/output-formats/_typst-citations.md:1]: This part could be simplified later when we'll "fix" our missing support for CSL that pandoc's support 

- https://github.com/quarto-dev/quarto-cli/issues/10168

This means also that Citation Style will apply.

**Review by cderv — APPROVED** [review]
> This adds the Typst details to the main citation page and calls out that Typst is different at the top. I think it still may be confusing since it isn't clear what parts of the page still apply to Typst (I think none, except "Citation Syntax" and I guess "Bibliography Files" although the supported formats might be different). Is it worth a bigger re-organize to clear that up, or is what is in this PR good enough.

I think it is good enough, but I understand the unclear state. Typst is I think the only format we have supporting citeproc but not defaulting to it. (Maybe we should change that 🤔 ) 

Probably a reorganization would be helpful. A simple one could be 

- This pages is about Citations. We could bigger sub part to clearly explain the different system that could happen
- One about Pandoc's citeproc built in system. Almost all content of the current page would be under it
- One about Format specific system and 
	- A subpart about LaTeX system
	- Another about typst own system

The pandoc syntax could be at the top as a common part. 

Not urgent though, it works well like this.

**Comment by cwickham** [comment]
@cderv Thanks for the feedback. I opened https://github.com/quarto-dev/quarto-cli/issues/10204 to track these suggestions and will merge this.

**Pull request merged**